### PR TITLE
new document explaining CLVM costs

### PR DESCRIPTION
**Pull request by danieljperry**
I added Arvid's original document with some formatting for markdown format.

### CODE REVIEW

**Comment by arvidn** [comment]
I think it's a bit risky (and not all that pertinent) to include the original benchmarks run in this document. They are old, and the `clvm_rs` implementation has advanced since then. But perhaps more importantly, I expect `clvm_rs` to be optimized further, to make these benchmarks more out-dated.

**Comment by mariano54** [comment]
It seems useful to have, so people can get a sense of things cost what they cost. What if we put disclaimers saying these are outdated and were used for the initial launch?

File: docs/06mempool/costs_benchmarking.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+The cost is approximately in units of nanoseconds, this means the cost for the BLS operations are 2303030000.
+
+## Max block cost
+
+This makes the max cost for (2) 1317054957 + 2303030000 = 3620084957. This is also the cost ascribed to (1) and (3).
+
+The max cost for the whole block is: 3620084957 * 3 = 10860254871.
+
+The cost for 1 `AGG_SIG_UNSAFE` and `AGG_SIG_ME` condition is 2303030000 / 2000 = 1151515.
```
Comment (arvidn) [inline docs/06mempool/costs_benchmarking.md:48]:
```suggestion
The cost for one `AGG_SIG_UNSAFE` and `AGG_SIG_ME` condition is 2303030000 / 2000 = 1151515.

```

**Review by arvidn — COMMENTED** [review]
I don't think the clvm benchmarks (i.e. the `concat.hex` et.al.) belong here. They are out-dated and don't mean anything without a lot more context of what those benchmark programs actually do. Especially in relation to the *cost* we ascribe to them.

Also, this document is not a detailed documentation of exactly how cost is ascribed to a clvm program. This document is the background work used to justify the costs we've ascribed to clvm operations. I think it would make more sense as an appendix.

File: docs/06mempool/costs_benchmarking.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+Each public key used in an aggregate signature validation itself needs to be validated, to ensure
+it is within the G1 subgroup. This cost is called "Public key validation" in the benchmarks.
+
+The validation of an aggregate signature takes all the public keys and the signature. The time it takes per public key is measured by the benchmark.
+
+For an `AGG_SIG_UNSAFE` and `AGG_SIG_ME` condition, we have to pay the cost for both the public key validation as well as the batch signature validation.
+
+## Intel MacBook Pro (MacOS):
+
+Public key validation<br/>
```
[the remark this answers is not in the document]
Comment (danieljperry) [reply docs/06mempool/costs_benchmarking.md:100]: Yes, and it in this case it's needed to avoid large gaps between lines.

**Comment by danieljperry** [comment]
Closing this PR as I have opened a new PR with the same info in the chialisp-web repo:
https://github.com/Chia-Network/chialisp-web/pull/163

**Pull request closed without merging**